In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import sklearn
from sklearn import model_selection, tree, metrics,svm
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split, GridSearchCV, KFold, cross_val_score
from sklearn.metrics import mean_squared_error, mean_squared_log_error, roc_auc_score
from sklearn.preprocessing import Normalizer, StandardScaler
from math import sqrt
import warnings
warnings.filterwarnings("ignore")

train = pd.read_excel("liedetector.xlsx")
train

X1       X2        X3       X4       X5  X6
0      P_001  3.62160   8.66610  -2.8073 -0.44699   0
1      P_002  4.54590   8.16740  -2.4586 -1.46210   0
2      P_003  3.86600  -2.63830   1.9242  0.10645   0
3      P_004  3.45660   9.52280  -4.0112 -3.59440   0
4      P_005  0.23230  -4.45520   4.5718 -0.98880   0
...      ...      ...       ...      ...      ...  ..
1367  P_1368  0.40614   1.34920  -1.4501 -0.55949   0
1368  P_1369 -1.38870  -4.87730   6.4774  0.34179   0
1369  P_1370 -3.75030 -13.45860  17.5932 -2.77710   0
1370  P_1371 -3.56370  -8.38270  12.3930 -1.28230   0
1371  P_1372 -2.54190  -0.65804   2.6842  1.19520   0

[1372 rows x 6 columns]

In [2]:
train.drop(['X1'], axis=1, inplace=True )

In [3]:
train

X2        X3       X4       X5  X6
0     3.62160   8.66610  -2.8073 -0.44699   0
1     4.54590   8.16740  -2.4586 -1.46210   0
2     3.86600  -2.63830   1.9242  0.10645   0
3     3.45660   9.52280  -4.0112 -3.59440   0
4     0.23230  -4.45520   4.5718 -0.98880   0
...       ...       ...      ...      ...  ..
1367  0.40614   1.34920  -1.4501 -0.55949   0
1368 -1.38870  -4.87730   6.4774  0.34179   0
1369 -3.75030 -13.45860  17.5932 -2.77710   0
1370 -3.56370  -8.38270  12.3930 -1.28230   0
1371 -2.54190  -0.65804   2.6842  1.19520   0

[1372 rows x 5 columns]

In [4]:
logistic = LogisticRegression()

In [5]:
X = train.drop('X6',axis=1)
y = train['X6']

In [6]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1372 entries, 0 to 1371
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   X2      1372 non-null   float64
 1   X3      1372 non-null   float64
 2   X4      1372 non-null   float64
 3   X5      1372 non-null   float64
 4   X6      1372 non-null   int64  
dtypes: float64(4), int64(1)
memory usage: 53.7 KB


In [7]:
logistic.fit(X,y)

LogisticRegression()

In [8]:
pred = logistic.predict(X)

In [9]:
metrics.confusion_matrix(y, pred)

array([[674, 140],
       [ 78, 480]])

### Ouestion 6. To answer the question, please follow the instructions given below: • Use the given dataset • Drop the "Prisoner Ids" Variable• Do not scale the data• Do not split the data into training and test. Build a Logistic Regression Model on the data set. What is the value of the True Positives predictions based on the confusion matrix? Note: For this question, use the whole data to build the model, then use the independent variables to predict the same whole data and then build a confusion matrix. Note: Please do not pre-process the data in any way before fitting the logistic regression model.

#### Option a is the answer (480)

In [15]:
metrics.roc_auc_score(y, logistic.predict_proba(X)[:, 1])

0.8517245251116219

### Question 7. To answer the question, please follow the instructions given below:

#### Option a is the answer (.852)